### ISSUE TITLE
Count with condition

### ISSUE THREAD

**Issue by ktibi** [infracost/infracost#2080]
Hello,

I have issue when I try to use infracost on a ressource with a count mixed with condition. Example with NAT gateway on AWS : 

`resource "aws_nat_gateway" "nat-gw" {
  count         = var.nat-mode == "none" ? 0 : var.nat-mode == "single" ? 1 : var.nat-mode == "multi" ? length(aws_subnet.public) : 0
`
Infracost say I'll deploy 6 nat GW module.network.aws_nat_gateway.nat-gw[1] ==> module.network.aws_nat_gateway.nat-gw[6]

I don't know if infracost can resolve this case with a configuration.

**Comment by tim775** [comment]
Hi, what version of infracost are you using?  

I made a similair tf config to what you posted but I haven't been able to reproduce the problem.  Could you provide a more complete example?  

This is what I tried that seems to work ok:

```
provider "aws" {
  region                      = "us-east-1"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true
  skip_get_ec2_platforms      = true
  skip_region_validation      = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
}

variable "nat-mode" {
  type    = string
  default = "single"
}

resource "aws_nat_gateway" "nat" {
  allocation_id = "eip-12345678"
  subnet_id     = "subnet-12345678"
  count         = var.nat-mode == "none" ? 0 : var.nat-mode == "single" ? 1 : var.nat-mode == "multi" ? length([1,2,3,4]) : 0
}
```

**Comment by robh007** [comment]
Also, I hadn't realised that EIPs no longer incur a charge unless they're not being used. So I'm assuming that's got something to do behaviour I'm seeing.

**Comment by vdmgolub** [comment]
@robh007 Hey! Thanks for the example! Yes, I've tried to run it with `INFRACOST_LOG_LEVEL=debug` and `--show-skipped` and CLI detects these IPs as free. However, running the CLI with the plan doesn't detect them being used. I wonder if the plan behavior is incorrect. I haven't used this module, so maybe you can tell me what behavior is correct in your opinion.

**Comment by robh007** [comment]
Hi @vdmgolub, I think the plan behaviour is correct as they are IPs that are being created stand alone & not necessarily being allocated or attached. The attachment occurs within the `nat_gateway` resource. So although EIPs are now free, you now appear to pay & fee for when they aren't attached to something. Which is the wording in the from the plan file. Sorry I seem to have hijacked this thread now with possibly a different issue.

**Comment by vdmgolub** [comment]
@robh007 Yep looks like a different issue specific to the EIPs. Could you please create a new one so we can track it separately? :)

**Comment by robh007** [comment]
I opened #2094 @vdmgolub
